### DISCUSSION TITLE
Free/full-text seach in folder structure

### DISCUSSION THREAD

**Discussion by maxCSA** [microsoft-search/pnp-modern-search#2929] [Q&A] [answered]
Hi all,

I would like to search in folder structure from the free text search field. My understanding is that the folder structure must be added to the full-text index, but I can't figure out how to do so.

Just to be clear, here is a folder structure:

| DirAAA
---| DirBBB
-----| File1
-----| File2
-----| File3

I want to find all 3 files if I perform a basic seach for "BBB" or even "AAA".

Can anyone point me in the right direction?

Thanks in advance

**Comment by maxCSA** [comment, marked as answer]
I couldn't configure the complete path to be full-text searchable, but I found this workaround filter : 
`( {searchTerms}* OR ParentLink:"{searchTerms}" )`

After many tries, I realized that I have to add a * after the first {seachTerms} otherwise the seach result was empty by default. It is the correct way to fix this?

Thanks

**Reply by wobba** [reply]
Don't see why you would need * for the inital part. And this one is better:

`{searchTerms} OR Parentlink:({searchTerms})`
